lemur/plugins/lemur_gcp/endpoints.py

```diff
@@ -150,25 +150,28 @@ def update_target_proxy_sni_certs(project_id, credentials, endpoint, old_cert, n
     if kind not in ("targethttpsproxy", "targetsslproxy") or endpoint.registry_type != "gcp":
         raise NotImplementedError()
     current_app.logger.info(f"Rotating SNI cert for endpoint {endpoint.name}")
-    old_cert_name = certificates.get_name(old_cert.body)
-    old_cert_self_link = certificates.get_self_link(project_id, old_cert_name)
     new_cert_name = certificates.get_name(new_cert.body)
     new_cert_self_link = certificates.get_self_link(project_id, new_cert_name)
     if kind == "targethttpsproxy":
         client = target_https_proxies.TargetHttpsProxiesClient(credentials=credentials)
         proxy = client.get(project=project_id, target_https_proxy=endpoint.name)
-        if old_cert_self_link not in proxy.ssl_certificates:
+        cert_to_delete = certificates.find_cert(project_id, credentials, old_cert, proxy.ssl_certificates)
+        if cert_to_delete and cert_to_delete not in proxy.ssl_certificates:
```

Comment:
Question: In which case can `cert_to_delete` be truthy and also not in `proxy.ssl_certificates`? If `cert_to_delete` is returned from `find_cert()` then that seems to imply it must exist in `proxy.ssl_certificates` right? Should this be checking if `cert_to_delete` is not in `proxy.ssl_certificates` instead.

The concern here would be that a conditional which never evaluates to true might suggests this code block isn't doing what we intend it to do but let me know if I'm misunderstanding this.

Reply:
Good catch, yeah `find_cert()` returning a truthy value implies it's already in the list. I will update.